models/ACC/ACC.jl

```diff
@@ -59,118 +58,147 @@ const v_set = 30.0  # ego car's set speed
     return dx
 end
 
-# We choose the controller with 5 hidden layers.
-controller = read_nnet_mat(@modelpath("ACC", "controller_5_20.mat");
-                           act_key="act_fcns");
-
-# ## Specifications
-#
-# The verification objective of this system is that given a scenario where both
-# cars are driving safely, the lead car suddenly slows down with
-# ``a_{lead} = -2``.
-# We want to check whether there is a collision in the following 5 seconds.
-# A control period of 0.1 seconds is used.
-#
-# Formally, the safety specification can be expressed as:
-# ```math
-#     D_{rel} = x_{lead} - x_{ego} ≥ D_{safe},
-# ```
-# where ``D_{safe} = D_{default} + T_{gap} * v_{ego}``, ``T_{gap} = 1.4`` sec,
-# and ``D_{default} = 10``.
+struct AccSpecification
+    D_default
+    d_rel
+    d_safe
+end
 
-# The uncertain initial conditions are chosen to be:
-#
-# - ``x_{lead}(0) ∈ [90,110], v_{lead}(0) ∈ [32,32.2], γ_{lead}(0) = γ_{ego}(0) = 0``
-# - ``v_{ego}(0) ∈ [30, 30.2], x_{ego} ∈ [10,11]``
-
-# The initial states according to the specification are:
-X₀ = Hyperrectangle(low=[90, 32, 0, 10, 30, 0], high=[110, 32.2, 0, 11, 30.2, 0]);
-U₀ = ZeroSet(1);
-
-# The system has 6 state variables and 1 control variable:
-vars_idx = Dict(:state_vars=>1:6, :control_vars=>7);
-ivp = @ivp(x' = ACC!(x), dim: 7, x(0) ∈ X₀ × U₀);
-
-period = 0.1;  # control period
-T = 5.0;  # time horizon
-
-# The controller input is $(v_{set}, T_{gap}, x[5], x[1] - x[4], x[2] - x[5])$.
-# To extract from the current state the last three inputs to the network we define a projection matrix $M$.
-
-M = zeros(3, 6);
-M[1, 5] = 1.0;
-M[2, 1] = 1.0;
-M[2, 4] = -1.0;
-M[3, 2] = 1.0;
-M[3, 5] = -1.0;
-function preprocess(X::LazySet)  # version for set computations
-    Y1 = Singleton([v_set, T_gap])
-    Y2 = linear_map(M, X)
-    return cartesian_product(Y1, Y2)
-end;
-function preprocess(X::AbstractVector)  # version for simulations
-    Y1 = [v_set, T_gap]
-    Y2 = M * X
-    return vcat(Y1, Y2)
-end;
-control_preprocessing = FunctionPreprocessing(preprocess);
-
-control_normalization = NoNormalization();
-
-prob = ControlledPlant(ivp, controller, vars_idx, period, control_normalization,
-                       control_preprocessing);
-
-# The specification can be interpreted as a half-space constraint:
-# $x[1] - x[4] - T_{gap} * x[5] ≥ D_{default}$.
-T_gap = 1.4;
-D_default = 10.0;
-d_rel = [1.0, 0, 0, -1, 0, 0, 0];
-d_safe = [0, 0, 0, 0, T_gap, 0, 0];
-d_prop = d_rel - d_safe;
+function ACC()
+    ## We choose the controller with 5 hidden layers.
+    controller = read_nnet_mat(@modelpath("ACC", "controller_5_20.mat");
+                               act_key="act_fcns")
+
+    ## ## Specifications
```

Comment:
I think it's better in terms of presentation, to keep as much normal text as possible. With this change, the `Specifications` text and the equations below will be displayed as comments, instead of being rendered in LaTeX. But to keep all the Julia code in one place it's possible to use the [Literate.script](https://fredrikekre.github.io/Literate.jl/v2/outputformats/#Literate.script) conversion.

Reply:
Good catch. I understood that `script` removes the comments completely, which is not what we want. I now moved the LaTeX part to the front.